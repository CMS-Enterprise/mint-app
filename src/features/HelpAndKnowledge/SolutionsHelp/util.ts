import { tObject } from 'utils/translation';

import { AboutConfigType } from './SolutionDetails/Solutions/Generic/About';
import { TimelineConfigType } from './SolutionDetails/Solutions/Generic/Timeline';
import { HelpSolutionBaseType, HelpSolutionsType } from './solutionsMap';

export const timelineTranslationUtil = (solutionKey: string) =>
  tObject<keyof TimelineConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.timeline`
  );

export const aboutTranslationUtil = (solutionKey: string) =>
  tObject<keyof AboutConfigType, any>(
    `helpAndKnowledge:solutions.${solutionKey}.about`
  );

export const convertSolutionObjectToArray = (
  solutionMap: HelpSolutionsType
): HelpSolutionBaseType[] =>
  Object.values(solutionMap).map(solution => solution);
