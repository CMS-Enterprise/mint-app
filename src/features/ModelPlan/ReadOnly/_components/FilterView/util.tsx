import React from 'react';
import { ModelViewFilter } from 'gql/generated/graphql';

export const filterGroupParams = [
  ModelViewFilter.CCW.toLowerCase(),
  ModelViewFilter.CMMI.toLowerCase(),
  ModelViewFilter.CBOSC.toLowerCase(),
  ModelViewFilter.DFSDM.toLowerCase(),
  ModelViewFilter.IPC.toLowerCase(),
  ModelViewFilter.IDDOC.toLowerCase(),
  ModelViewFilter.MDM.toLowerCase(),
  ModelViewFilter.OACT.toLowerCase(),
  ModelViewFilter.PBG.toLowerCase()
];

type GroupOption = {
  value: (typeof filterGroupParams)[number];
  label: string;
};

export const groupOptions: GroupOption[] = [
  { value: 'ccw', label: 'Chronic Conditions Warehouse (CCW)' },
  { value: 'cmmi', label: 'CMMI Cost Estimate' },
  {
    value: 'cbosc',
    label: 'Consolidated Business Operations Support Center (CBOSC)'
  },
  {
    value: 'dfsdm',
    label: 'Division of Financial Services and Debt Management (DFSDM)'
  },
  { value: 'ipc', label: 'Innovation Payment Contractor (IPC)' },
  {
    value: 'iddoc',
    label: 'Innovative Design, Development, and Operations Contract (IDDOC)'
  },
  { value: 'mdm', label: 'Master Data Management (MDM)' },
  { value: 'oact', label: 'Office of the Actuary (OACT)' },
  { value: 'pbg', label: 'Provider Billing Group (PBG)' }
];

export const checkGroupMap = (
  isViewingFilteredView: boolean | undefined,
  filteredQuestions: string[] | undefined,
  question: string,
  component: React.ReactNode
) => {
  // Show the question if it is included in the map
  if (isViewingFilteredView && filteredQuestions?.includes(question)) {
    return component;
  }
  // Hide the question if it is NOT included in the map
  if (isViewingFilteredView && !filteredQuestions?.includes(question)) {
    return <></>;
  }
  // Return the component if it is NOT isViewingFilteredView
  return component;
};

export const highLevelTimelineQuestions = [
  'completeICIP',
  'clearanceStarts',
  'announced',
  'applicationsStart',
  'performancePeriodStarts',
  'wrapUpEnds'
];

export const hasQuestions = (
  allQuestions: string[],
  questionsInQuestion: string[]
) => {
  if (
    allQuestions?.filter(question => questionsInQuestion.includes(question))
      .length === 0
  ) {
    return false;
  }
  return true;
};
