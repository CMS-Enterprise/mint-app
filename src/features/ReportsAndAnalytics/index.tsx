import React, { useMemo, useState } from 'react';
import { Trans, useTranslation } from 'react-i18next';
import {
  Button,
  Card,
  CardBody,
  CardFooter,
  CardGroup,
  CardHeader,
  Grid,
  GridContainer,
  Header,
  Link,
  PrimaryNav,
  Select
} from '@trussworks/react-uswds';
import classNames from 'classnames';
import downloadAnalytics, {
  analyticsSummaryConfig,
  downloadChartAsPDF,
  downloadMTOMilestoneSummary,
  downloadMultipleChartsAsPDF,
  getChangesByOtherData,
  getChangesBySection,
  UsedAnalyticsSummaryKey
} from 'features/ReportsAndAnalytics/util';
import {
  TableName,
  useGetAnalyticsSummaryQuery,
  useGetMtoMilestoneSummaryQuery
} from 'gql/generated/graphql';
import {
  Bar,
  BarChart,
  CartesianGrid,
  Line,
  LineChart,
  ResponsiveContainer,
  Tooltip,
  XAxis,
  YAxis
} from 'recharts';

import CheckboxField from 'components/CheckboxField';
import MainContent from 'components/MainContent';
import PageLoading from 'components/PageLoading';
import Spinner from 'components/Spinner';
import useCheckResponsiveScreen from 'hooks/useCheckMobile';
import useFetchCSVData from 'hooks/useFetchCSVData';
import { reports } from 'i18n/en-US/analytics';
import tables from 'i18n/en-US/modelPlan/tables';
import { formatDateUtc } from 'utils/date';

export type ReportsType = 'mtoMilestoneSummary' | 'allModels';

const ReportsAndAnalytics = () => {
  const { t: modelPlanT } = useTranslation('modelPlan');
  const { t } = useTranslation('analytics');

  const isTablet = useCheckResponsiveScreen('tablet', 'smaller');
  const isMobile = useCheckResponsiveScreen('mobile', 'smaller');

  // Responsive margins and height for the chart
  const chartMargins = useMemo(() => {
    if (isMobile) {
      return { top: 40, right: 120, left: 40, bottom: 100 };
    }
    if (isTablet) {
      return { top: 40, right: 140, left: 80, bottom: 150 };
    }
    return { top: 40, right: 160, left: 120, bottom: 200 };
  }, [isMobile, isTablet]);

  const chartHeight = useMemo(() => {
    if (isMobile) return 400;
    if (isTablet) return 600;
    return 1000;
  }, [isMobile, isTablet]);

  // Format X-axis labels for date-based charts
  const formatXAxisLabel = (value: any) => {
    if (selectedChart === 'numberOfModelsOverTime') {
      // Format the monthYear value (e.g., "2023-01-01T00:00:00Z") to a readable date
      return formatDateUtc(value, 'MMMM yyyy');
    }
    return value;
  };

  const [selectedChart, setSelectedChart] = useState<string>('changesPerModel');

  const [appendTableToChart, setAppendTableToChart] = useState<boolean>(true);

  const [isDownloadingAllCharts, setIsDownloadingAllCharts] =
    useState<boolean>(false);

  // Fetch all data for CSV export of all model plans
  const { fetchAllData } = useFetchCSVData();

  const { data, loading, error } = useGetAnalyticsSummaryQuery({
    fetchPolicy: 'network-only' // Don't use cache for analytics summary
  });

  const analyticsData = data?.analytics || ({} as any);

  const { data: mtoMilestoneSummary, loading: mtoMilestoneSummaryLoading } =
    useGetMtoMilestoneSummaryQuery();

  const mtoMilestoneSummaryData = useMemo(() => {
    return mtoMilestoneSummary?.modelPlanCollection ?? [];
  }, [mtoMilestoneSummary?.modelPlanCollection]);

  // Some data is not an array, so we need to handle that
  let chartData: any = !Array.isArray(
    analyticsData[selectedChart as UsedAnalyticsSummaryKey]
  )
    ? [analyticsData[selectedChart as UsedAnalyticsSummaryKey]]
    : analyticsData[selectedChart as UsedAnalyticsSummaryKey];

  // Custom logic for changes per model by section and other data
  if (selectedChart === 'changesPerModelBySection') {
    chartData = getChangesBySection(analyticsData.changesPerModelBySection);
  } else if (selectedChart === 'changesPerModelOtherData') {
    chartData = getChangesByOtherData(analyticsData.changesPerModelOtherData);
  }

  // Onclick handler for downloading multiple charts as PDF
  const downloadMultipleChartsPDF = async () => {
    setIsDownloadingAllCharts(true);
    const originalChart = selectedChart; // Store the original chart
    const chartTypes = Object.keys(analyticsSummaryConfig);
    const singleChartData =
      analyticsData[selectedChart as UsedAnalyticsSummaryKey];
    const chartDataForPDF = Array.isArray(singleChartData)
      ? singleChartData
      : [];
    await downloadMultipleChartsAsPDF(
      chartTypes,
      'MINT-Analytics-All-Charts.pdf',
      setSelectedChart,
      chartDataForPDF
    );
    setIsDownloadingAllCharts(false);
    setSelectedChart(originalChart); // Restore the orignal chart
  };

  // If loading any data, show loading spinner
  if (loading || mtoMilestoneSummaryLoading)
    return <PageLoading data-testid="analytics-loading" />;

  return (
    <MainContent className="mint-body-normal">
      <GridContainer>
        <h1 className="margin-top-10 margin-bottom-2">{t('heading')}</h1>

        <p
          className="mint-body-large margin-top-0 margin-bottom-1"
          data-testid="analytics-description"
        >
          {t('description')}
        </p>

        <Trans
          i18nKey="analytics:contactMINTTeam"
          components={{
            email: <Link href="mailto:MINTTeam@cms.hhs.gov"> </Link>
          }}
        />

        <h2 className="margin-top-6 margin-bottom-2">
          {t('downloadableReports')}
        </h2>

        <p className="mint-body-normal margin-top-0">
          {t('downloadableReportsDescription')}
        </p>

        <CardGroup className="padding-x-1 margin-y-4">
          <Grid desktop={{ col: 12 }}>
            <Grid row gap={1}>
              {Object.keys(reports).map(reportKey => (
                <Grid desktop={{ col: 4 }} tablet={{ col: 6 }} key={reportKey}>
                  <Card
                    containerProps={{
                      className: 'radius-md padding-0 margin-0',
                      style: {
                        minHeight: '350px'
                      }
                    }}
                    className="margin-bottom-2"
                  >
                    <CardHeader className="padding-3 padding-bottom-0">
                      <h3 className="line-height-normal margin-top-1">
                        {t(reports[reportKey as ReportsType].heading)}
                      </h3>
                    </CardHeader>

                    <CardBody className="padding-x-3 ">
                      <p>{t(reports[reportKey as ReportsType].description)}</p>

                      <p className="text-base-dark">
                        {t(reports[reportKey as ReportsType].formatExcel)}
                      </p>
                    </CardBody>

                    <CardFooter className="padding-3">
                      <Button
                        type="button"
                        className="margin-right-2"
                        disabled={mtoMilestoneSummaryLoading}
                        data-testid={`download-${reportKey}-button`}
                        onClick={() => {
                          if (reportKey === 'mtoMilestoneSummary') {
                            downloadMTOMilestoneSummary(
                              mtoMilestoneSummaryData,
                              'MINT-MTO_Milestone_Summary.xlsx'
                            );
                          } else if (reportKey === 'allModels') {
                            fetchAllData();
                          }
                        }}
                      >
                        {t('download')}
                      </Button>
                    </CardFooter>
                  </Card>
                </Grid>
              ))}
            </Grid>
          </Grid>
        </CardGroup>

        <h2 className="margin-top-6 margin-bottom-2">{t('mintAnalytics')}</h2>

        <p className="mint-body-normal margin-top-0">
          {t('mintAnalyticsDescription')}
        </p>

        {!analyticsData || error ? (
          <p data-testid="no-analytics-data">{t('noAnalyticsData')}</p>
        ) : (
          <>
            <Header
              basic
              extended={false}
              className={classNames(
                'model-to-operations__nav-container margin-top-4',
                {
                  'border-bottom border-base-lighter': !isTablet
                }
              )}
              data-testid="chart-navigation"
            >
              <div className="usa-nav-container padding-0">
                <PrimaryNav
                  items={Object.keys(analyticsSummaryConfig).map(item => (
                    <Button
                      type="button"
                      onClick={() => {
                        setSelectedChart(item);
                      }}
                      key={item}
                      data-testid={`chart-tab-${item}`}
                      style={{
                        textAlign: 'center'
                      }}
                      className={classNames(
                        'usa-nav__link margin-left-neg-2 margin-right-2',
                        {
                          'usa-current': selectedChart === item
                        }
                      )}
                    >
                      <span
                        className={classNames({
                          'text-primary': selectedChart === item
                        })}
                      >
                        {t(item)}
                      </span>
                    </Button>
                  ))}
                  mobileExpanded={false}
                  className="flex-justify-start margin-0 padding-0"
                />
              </div>
            </Header>

            {isTablet && (
              <div className="maxw-mobile-lg">
                <p className="margin-y-0 text-bold">{t('report')}</p>
                <Select
                  id="selected-chart"
                  name="selectedChart"
                  value={selectedChart}
                  data-testid="mobile-chart-selector"
                  onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
                    setSelectedChart(e.target.value);
                  }}
                  className="margin-bottom-0 text-primary text-bold margin-top-1"
                >
                  {Object.keys(analyticsSummaryConfig).map(item => {
                    return (
                      <option key={item} value={item}>
                        {t(item)}
                      </option>
                    );
                  })}
                </Select>
              </div>
            )}

            {/* <div> */}
            <Button
              type="button"
              className="margin-top-4 margin-right-2"
              unstyled
              data-testid="download-analytics-button"
              onClick={() => {
                downloadAnalytics(analyticsData, 'MINT-Analytics.xlsx');
              }}
            >
              {t('downloadExcel')}
            </Button>

            <Button
              type="button"
              className="margin-top-1 margin-right-2"
              unstyled
              data-testid="download-chart-pdf-button"
              onClick={() => {
                downloadChartAsPDF(
                  `analytics-chart-${selectedChart}`,
                  'MINT-Analytics-Chart.pdf',
                  chartData,
                  selectedChart
                );
              }}
            >
              {t('downloadPDF')}
            </Button>

            <Button
              type="button"
              className="margin-top-1 margin-right-4"
              unstyled
              data-testid="download-multiple-charts-pdf-button"
              disabled={isDownloadingAllCharts}
              onClick={downloadMultipleChartsPDF}
            >
              {t('downloadMultipleChartsPDF')}
              {isDownloadingAllCharts && (
                <Spinner size="small" className="margin-right-1" />
              )}
            </Button>

            <div className="display-inline-block">
              <CheckboxField
                id="append-table-to-chart"
                name="appendTableToChart"
                onBlur={() => {}}
                value="true"
                label={t('appendTableToChart')}
                checked={appendTableToChart}
                onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
                  setAppendTableToChart(e.target.checked);
                }}
              />
            </div>
            {/* </div> */}

            <ResponsiveContainer
              width="100%"
              height={chartHeight}
              data-testid="chart-container"
              id={`analytics-chart-${selectedChart}`}
            >
              {analyticsSummaryConfig[selectedChart as UsedAnalyticsSummaryKey]
                .chartType === 'bar' ? (
                <BarChart data={chartData as any[]} margin={chartMargins}>
                  <CartesianGrid strokeDasharray="3 3" />

                  <XAxis
                    dataKey={
                      analyticsSummaryConfig[
                        selectedChart as UsedAnalyticsSummaryKey
                      ].xAxisDataKey
                    }
                    angle={-45}
                    textAnchor="end"
                    height={80}
                    tickFormatter={value => {
                      // Format different types of X-axis labels
                      if (selectedChart === 'modelsByStatus') {
                        // Format status values (e.g., "ACTIVE" -> "Active")
                        return modelPlanT(`status.options.${value}`);
                      }
                      if (
                        selectedChart === 'changesPerModelBySection' ||
                        selectedChart === 'changesPerModelOtherData'
                      ) {
                        // Format table names (e.g., "plan_basics" -> "Plan Basics")
                        return tables[value as TableName].generalName;
                      }
                      // For model names and other text, truncate if too long
                      if (typeof value === 'string' && value.length > 15) {
                        return `${value.substring(0, 12)}...`;
                      }
                      return value;
                    }}
                  />
                  <YAxis />

                  <Tooltip
                    formatter={(value, name) => [
                      value,
                      t(name as UsedAnalyticsSummaryKey)
                    ]}
                  />

                  <Bar
                    dataKey={
                      analyticsSummaryConfig[
                        selectedChart as UsedAnalyticsSummaryKey
                      ].yAxisDataKey
                    }
                    fill="#008480"
                    animationDuration={300}
                    animationEasing="ease-out"
                  />
                </BarChart>
              ) : (
                <LineChart data={chartData as any[]} margin={chartMargins}>
                  <CartesianGrid strokeDasharray="3 3" />
                  <XAxis
                    dataKey={
                      analyticsSummaryConfig[
                        selectedChart as UsedAnalyticsSummaryKey
                      ].xAxisDataKey
                    }
                    tickFormatter={formatXAxisLabel}
                  />
                  <YAxis />

                  <Tooltip
                    formatter={(value, name, props: any) => {
                      // Format the tooltip based on chart type
                      if (selectedChart === 'numberOfModelsOverTime') {
                        return [value, `${t(name as UsedAnalyticsSummaryKey)}`];
                      }
                      return [value, t(name as UsedAnalyticsSummaryKey)];
                    }}
                    labelFormatter={label => {
                      // Format the label (X-axis value) for date-based charts
                      if (selectedChart === 'numberOfModelsOverTime') {
                        return formatDateUtc(label, 'MMMM yyyy');
                      }
                      return label;
                    }}
                  />

                  <Line
                    dataKey={
                      analyticsSummaryConfig[
                        selectedChart as UsedAnalyticsSummaryKey
                      ].yAxisDataKey
                    }
                    stroke="#008480"
                    strokeWidth={3}
                    animationDuration={300}
                    animationEasing="ease-out"
                    dot={{ r: 4 }}
                    activeDot={{ r: 6 }}
                  />
                </LineChart>
              )}
            </ResponsiveContainer>
          </>
        )}
      </GridContainer>
    </MainContent>
  );
};

export default ReportsAndAnalytics;
