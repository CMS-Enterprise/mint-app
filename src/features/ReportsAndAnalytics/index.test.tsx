import React from 'react';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitFor } from '@testing-library/react';
import { analyticsSummaryMock, mockAnalyticsData } from 'tests/mock/general';
import { mtoMilestoneSummaryMock } from 'tests/mock/mto';
import { expect, it } from 'vitest';

import { getChangesByOtherData, getChangesBySection } from './util';
import ReportsAndAnalytics from '.';

// Mock ResizeObserver for ResponsiveContainer
global.ResizeObserver = vi.fn().mockImplementation(() => ({
  observe: vi.fn(),
  unobserve: vi.fn(),
  disconnect: vi.fn()
}));

describe('ReportsAndAnalytics', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  const router = createMemoryRouter(
    [
      {
        path: '/reports-and-analytics',
        element: <ReportsAndAnalytics />
      }
    ],
    {
      initialEntries: ['/reports-and-analytics']
    }
  );

  it('matches the snapshot when analytics feature is enabled and data is loaded', async () => {
    const { asFragment } = render(
      <MockedProvider
        mocks={[...analyticsSummaryMock, ...mtoMilestoneSummaryMock]}
        addTypename={false}
      >
        <RouterProvider router={router} />
      </MockedProvider>
    );

    await waitFor(() => {
      expect(screen.getByText('Changes per model')).toBeInTheDocument();
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('returns cumulative changes by section', async () => {
    const expected = getChangesBySection(
      mockAnalyticsData.analytics.changesPerModelBySection
    );

    expect(expected).toEqual([
      { tableName: 'plan_basics', numberOfChanges: 8 }
    ]);
  });

  it('returns cumulative changes by other data', async () => {
    const expected = getChangesByOtherData(
      mockAnalyticsData.analytics.changesPerModelOtherData
    );

    expect(expected).toEqual([
      { section: 'plan_documents', numberOfChanges: 5 }
    ]);
  });
});
