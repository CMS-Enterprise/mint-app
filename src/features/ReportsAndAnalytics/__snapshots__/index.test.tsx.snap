// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ReportsAndAnalytics > matches the snapshot when analytics feature is enabled and data is loaded 1`] = `
<DocumentFragment>
  <main
    class="mint-main-content mint-body-normal"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <div
      class="grid-container"
      data-testid="gridContainer"
    >
      <h1
        class="margin-top-10 margin-bottom-2"
      >
        Reports and analytics
      </h1>
      <p
        class="mint-body-large margin-top-0 margin-bottom-1"
        data-testid="analytics-description"
      >
        Download reports containing a variety of information about models in MINT or view the latest analytics about content in MINT.
      </p>
      <a
        class="usa-link"
        href="mailto:MINTTeam@cms.hhs.gov"
      >
        Contact the MINT Team
      </a>
       if you have any questions about the available reports.
      <h2
        class="margin-top-6 margin-bottom-2"
      >
        Downloadable reports
      </h2>
      <p
        class="mint-body-normal margin-top-0"
      >
        Download reports of the latest model information in MINT. Each report will be downloaded in a specific format and will include a subset of model information available in MINT.
      </p>
      <ul
        class="usa-card-group padding-x-1 margin-y-4"
        data-testid="CardGroup"
      >
        <div
          class="desktop:grid-col-12"
          data-testid="grid"
        >
          <div
            class="grid-row grid-gap-1"
            data-testid="grid"
          >
            <div
              class="tablet:grid-col-6 desktop:grid-col-4"
              data-testid="grid"
            >
              <li
                class="usa-card margin-bottom-2"
                data-testid="Card"
              >
                <div
                  class="usa-card__container radius-md padding-0 margin-0"
                  style="min-height: 350px;"
                >
                  <div
                    class="usa-card__header padding-3 padding-bottom-0"
                    data-testid="CardHeader"
                  >
                    <h3
                      class="line-height-normal margin-top-1"
                    >
                      MTO milestone summary
                    </h3>
                  </div>
                  <div
                    class="usa-card__body padding-x-3 "
                    data-testid="CardBody"
                  >
                    <p>
                      This report downloads an excel file of all model-to-operations matrix (MTO) milestones, separated by model. It contains a subset of data about each milestone.
                    </p>
                    <p
                      class="text-base-dark"
                    >
                      Format: Excel
                    </p>
                  </div>
                  <div
                    class="usa-card__footer padding-3"
                    data-testid="CardFooter"
                  >
                    <button
                      class="usa-button margin-right-2"
                      data-testid="download-mtoMilestoneSummary-button"
                      type="button"
                    >
                      Download
                    </button>
                  </div>
                </div>
              </li>
            </div>
            <div
              class="tablet:grid-col-6 desktop:grid-col-4"
              data-testid="grid"
            >
              <li
                class="usa-card margin-bottom-2"
                data-testid="Card"
              >
                <div
                  class="usa-card__container radius-md padding-0 margin-0"
                  style="min-height: 350px;"
                >
                  <div
                    class="usa-card__header padding-3 padding-bottom-0"
                    data-testid="CardHeader"
                  >
                    <h3
                      class="line-height-normal margin-top-1"
                    >
                      All models
                    </h3>
                  </div>
                  <div
                    class="usa-card__body padding-x-3 "
                    data-testid="CardBody"
                  >
                    <p>
                      This report downloads a csv of information about all models in MINT including Model Plan details and more.
                    </p>
                    <p
                      class="text-base-dark"
                    >
                      Format: CSV
                    </p>
                  </div>
                  <div
                    class="usa-card__footer padding-3"
                    data-testid="CardFooter"
                  >
                    <button
                      class="usa-button margin-right-2"
                      data-testid="download-allModels-button"
                      type="button"
                    >
                      Download
                    </button>
                  </div>
                </div>
              </li>
            </div>
          </div>
        </div>
      </ul>
      <h2
        class="margin-top-6 margin-bottom-2"
      >
        MINT analytics
      </h2>
      <p
        class="mint-body-normal margin-top-0"
      >
        Charts and graphs of current statistics for MINT.
      </p>
      <header
        class="usa-header usa-header--basic model-to-operations__nav-container margin-top-4 border-bottom border-base-lighter"
        data-testid="chart-navigation"
      >
        <div
          class="usa-nav-container padding-0"
        >
          <nav
            class="usa-nav flex-justify-start margin-0 padding-0"
          >
            <button
              aria-label="Close Navigation Menu"
              class="usa-nav__close"
              data-testid="navCloseButton"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="usa-icon usa-icon--size-3"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
            <ul
              class="usa-nav__primary usa-accordion"
            >
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2 usa-current"
                  data-testid="chart-tab-changesPerModel"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class="text-primary"
                  >
                    Changes per model
                  </span>
                </button>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2"
                  data-testid="chart-tab-changesPerModelBySection"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class=""
                  >
                    Changes per model by section
                  </span>
                </button>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2"
                  data-testid="chart-tab-changesPerModelOtherData"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class=""
                  >
                    Changes per model other data
                  </span>
                </button>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2"
                  data-testid="chart-tab-modelsByStatus"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class=""
                  >
                    Models by status
                  </span>
                </button>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2"
                  data-testid="chart-tab-numberOfFollowersPerModel"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class=""
                  >
                    Number of followers per model
                  </span>
                </button>
              </li>
              <li
                class="usa-nav__primary-item"
              >
                <button
                  class="usa-button usa-nav__link margin-left-neg-2 margin-right-2"
                  data-testid="chart-tab-numberOfModelsOverTime"
                  style="text-align: center;"
                  type="button"
                >
                  <span
                    class=""
                  >
                    Number of models over time
                  </span>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <button
        class="usa-button usa-button--unstyled margin-top-4 margin-right-2"
        data-testid="download-analytics-button"
        type="button"
      >
        Download data (excel)
      </button>
      <button
        class="usa-button usa-button--unstyled margin-top-1 margin-right-2"
        data-testid="download-chart-pdf-button"
        type="button"
      >
        Download chart (PDF)
      </button>
      <button
        class="usa-button usa-button--unstyled margin-top-1 margin-right-2"
        data-testid="download-multiple-charts-pdf-button"
        type="button"
      >
        Download all charts (PDF)
      </button>
      <div
        class="recharts-responsive-container"
        id="analytics-chart-changesPerModel"
        style="width: 100%; height: 1000px; min-width: 0;"
      >
        <div
          style="width: 0px; height: 0px; overflow: visible;"
        />
      </div>
    </div>
  </main>
</DocumentFragment>
`;
